app/src/main/kotlin/Reversi/ReversiSolver.kt

```diff
@@ -1,22 +1,48 @@
 package Reversi
 
+import StateNode
 import StatefulSolver
-import StatelessSolver
 import java.awt.Point
-import kotlin.random.Random
-
-class ReversiSolver(private val initialState: ReversiState) {
-    fun getMove(state: ReversiState) : Point {
-        val game = ReversiMDP(state)
-        val solver = StatelessSolver(
-            game,
-            2000,
-            1.4,
-            0.9,
-            true
-        )
-
-        solver.constructTree(60)
-        return solver.extractOptimalAction()
+
+class ReversiSolver(initialState: ReversiState)
+    : StatefulSolver<ReversiState, Point>(ReversiMDP(initialState), 2000, 1.4, 0.9, false) {
+
+    override fun iterateStep() {
+//        displayTree(4)
+        super.iterateStep()
+    }
+
+    override fun selectNode(node: StateNode<ReversiState, Point>): StateNode<ReversiState, Point> {
+        val adversarial = adversarialSelect(node)
+        val greedy = super.selectNode(adversarial)
+
+        return greedy
+    }
+
+    private fun adversarialSelect(node: StateNode<ReversiState, Point>): StateNode<ReversiState, Point> {
```

Comment:
As we discussed last meeting, this is a custom search you designed just for Reversi. Perhaps we may base this search on something SoTA Reversi AI's actually use? Perhaps a bit more literature review on this subject too. But not a big deal, just as discussed, we should benchmark our class heuristic extension with Vanilla MCTS - that is sufficient for our use case.